### PR TITLE
223- Update User API Call

### PR DESCRIPTION
**Pull request by austin-bmitchell15**
Updated the API call to remove the several queries to instead send a single object in the body of the request. Currently, it works as a refactor to simplify a lot of code. Also Colin made me delete the Settings directory. So if prod breaks it is his fault.

223-update-user-api-rewrite

Code rewrites how both updateUser is done also consolidates a lot of code to remove old API calls like updateRole() and updateProfile()

ISSUE: #223 

#### Dev Checklist
- [x] If there was a database change, did you create an upgrade script.
- [x] Did you link the related github issue(s)?
- [x] Did you list any known issues/questions with the PR?

### CODE REVIEW

File: src/utils/user.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface User {
```
Comment (colin4554) [inline src/utils/user.ts]: Did we end up using this? If not, you can delete it

File: src/actions/queries.js [hunk 2]
```diff
@@ -12,52 +12,8 @@ export const getCurrentUser = (userId) =>
...
-
-  if (query.length > 0) {
-    query = query.slice(0, -1);
-    axios.post(`/api/users/updateUser?email=${email}&${query}`, { userId });
-  }
+export const updateUser = (id, user, role) => {
+  axios.post(`/api/users/${id}`, user, role);;
 };
 
 export const updateRole = (email, role) =>
```
Comment (colin4554) [inline src/actions/queries.js]: Please delete this function too

File: src/actions/queries.js [hunk 3]
```diff
@@ -12,52 +12,8 @@ export const getCurrentUser = (userId) =>
...
-  if (notes) {
-    query += "notes=" + notes + "&";
-  }
-
-  if (query.length > 0) {
-    query = query.slice(0, -1);
-    axios.post(`/api/users/updateUser?email=${email}&${query}`, { userId });
-  }
+export const updateUser = (id, user, role) => {
+  axios.post(`/api/users/${id}`, user, role);;
```
Comment (colin4554) [inline src/actions/queries.js]: Why do you have role as a parameter here?

Comment (colin4554) [reply src/actions/queries.js]: Currently, the role isn't being updated correctly. Can you take a look?

File: server/actions/users.js [hunk 4]
```diff
@@ -168,193 +168,24 @@ export async function getCurrentUser(userId, next) {
...
-  last_name,
-  date_of_birth,
-  zip_code,
-  address,
-  city,
-  state,
-  notes,
-  userId
-) {
+export async function updateUser(id, user) {
```
Comment (colin4554) [inline server/actions/users.js]: Careful with this function. The `userId` was actually referring to the adminId of whoever made the change. The createHistoryEventEditProfile keeps an internal record of any admin changes. Sorry, I forgot to mention this, but it looks like you will need to pass an adminId, the userId for the user being updated, and an object that holds the updates.

**Review by colin4554 — CHANGES_REQUESTED** [review]

File: src/actions/queries.js (hunk 3 above)
Comment (austin-bmitchell15) [reply src/actions/queries.js]: Yeah, I am looking into it now

Comment (austin-bmitchell15) [reply src/actions/queries.js]: This should be resolved

**Comment by YiQin93** [comment]
The profile edit and assistant editing work perfectly!! 

There seems to be an issue with editing a volunteer's profile from the volunteer list: the volunteer's profile edits got saved immediately after clicking "update" button, but the edits disappeared after refreshing the page. Please see this report for more details: https://go.playerzero.app/share/63decf22e70f67996c5c2f7e

**Comment by YiQin93** [comment]
This looks good!! Thanks Austin for fixing it right away!

**Review by colin4554 — APPROVED** [review]

**Pull request merged**